dart/lib/flat_buffers.dart

```diff
@@ -718,11 +727,7 @@ class Builder {
         int deltaCapacity = desiredNewCapacity - oldCapacity;
         deltaCapacity += (-deltaCapacity) % _maxAlign;
         int newCapacity = oldCapacity + deltaCapacity;
-        ByteData newBuf = new ByteData(newCapacity);
-        newBuf.buffer
-            .asUint8List()
-            .setAll(deltaCapacity, _buf.buffer.asUint8List());
-        _buf = newBuf;
+        _buf = _allocator.reallocateDownward(_buf, newCapacity, oldCapacity, 0);
```

Comment:
nit: I think `resize` is more idiomatic.

We also might want to make this a method on the buffer itself, and instead of making `_buf` direclty be a `ByteData` we make it some interface type that can be implemented by ByteData.

Reply:
Sure thing, I'll rename. The name's comming from the c++ allocator.

About making this part of the `_buf` - yeah, I've thought about it but it would make implementing an allocator a little more involved so I thought I'd follow the example given in the c++ allocator: https://github.com/google/flatbuffers/blob/master/include/flatbuffers/flatbuffers.h#L655